src/main/java/com/zufar/onlinestore/user/api/GetPasswordByEmail.java

```diff
@@ -0,0 +1,21 @@
+package com.zufar.onlinestore.user.api;
+
+import com.zufar.onlinestore.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class GetPasswordByEmail {
+
+    private final UserRepository userRepository;
+
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.READ_COMMITTED, readOnly = true)
+    public String getPasswordByEmail(String email) {
+        return userRepository.findPasswordByEmail(email)
+                .orElseThrow(() -> new RuntimeException("User not found"));
```

Comment:
please be more concise on how we should handle the case when password cannot be found by email. Throw specific exception ( not runtime ).